app/models/spotlight/resource.rb

```diff
@@ -40,7 +41,13 @@ def needs_provider?
     #
     # @param [Hash] All arguments will be passed through to ActiveRecord's #save method
     def save_and_index(*args)
-      save(*args) && reindex
+      save(*args) && reindex_later
```

Comment:
Yep. This uses the same `ActiveJob` step that we're using when you click the "Reindex" button.